hell/adventcalendar/2025/5/index.md

```diff
@@ -1,35 +1,309 @@
+
+### Complex relationships
+
+ARIA also has many attributes that help express more complex relationships between elements: `aria-controls`, `aria-owns`, `aria-activedescendant` and `aria-flowto`.
+
+`aria-controls` is used to indicate that an element controls another element, for example a button that shows or hides a section of content. This can help assistive technologies understand the relationship between the button and the content it controls:
+
+```html
+<button aria-controls="submenu" aria-expaned="false">Open menu</button>
+<nav id="submenu" hidden>...</nav>
+```
+
+Here, the button is indicating that it controls the visibility of the `nav` element with the ID `submenu`. You can then combine this with `aria-expanded` on the button to indicate whether the submenu is currently visible or hidden.
+
+To account for even more complex structures, `aria-controls` can also reference multiple IDs, allowing a single control to manage several elements at once.
+
+`aria-owns` is used to create a parent-child relationship between elements that are not nested in the DOM, think of a chat widget button and the chat window that opens when you click it. The button and chat window might be siblings in the DOM, but you can use `aria-owns` to indicate that the button "owns" the chat window:
+
+```html
+<button aria-owns="chat-window">Open chat</button>
+<div id="chat-window" hidden>...</div>
+```
+
+This helps assistive technologies understand the relationship between the button and the chat window, even though they are not nested. Like `aria-controls`, `aria-owns` can also reference multiple IDs to establish ownership over several elements.
+
+The difference between `aria-owns` and `aria-controls` is subtle: `aria-owns` is used to indicate a specific hierarchy, While `aria-controls` indicates a functional relationship.
+
+`aria-activedescendant` is used to indicate which element within a _composite widget_ is currently active. For example, in a combobox, which is like a select that also lets you type in it (or: a combination of an text input and a listbox), you can use `aria-activedescendant` (in combination with `aria-control`) on the input to indicate which option in the listbox is currently selected:
+
+```html
+<input
+  type="text"
+  aria-activedescendant="option-2"
+  aria-controls="options-list"
+/>
+<ul id="options-list" role="listbox">
+  <li id="option-1" role="option">Option 1</li>
+  <li id="option-2" role="option">Option 2</li>
+  <li id="option-3" role="option">Option 3</li>
+</ul>
+```
+
+Lastly, `aria-flowto` is used to indicate a logical reading order between elements that doesn't follow the visual ordering of elements, which can happen when using the `order` CSS property to change the visual order, or when you position elements with `position: absolute`. Using `aria-flowto` can help assistive technologies navigate the content in a way that makes sense.
```

Comment:
Please mention that the correct order should be established in the document and something like aria-flowto should be considered a last resort. Also, please mention that browser/sr support isn't great.

https://a11ysupport.io/tech/aria/aria-flowto_attribute